shared/rangeanalysis/codeql/rangeanalysis/RangeAnalysis.qll

```diff
@@ -160,17 +160,62 @@ signature module Semantic {
   /** Gets a tiebreaker id in case `getBlockId1` is not unique. */
   default string getBlockId2(BasicBlock bb) { result = "" }
 
+  /**
+   * A guard in the range analysis.
+   */
   class Guard {
+    /**
+     * Gets a string representation of the guard.
+     */
     string toString();
 
+    /**
+     * Gets the basic block associated with the guard.
+     */
     BasicBlock getBasicBlock();
 
+    /**
+     * Gets the guard as an expression.
+     */
     Expr asExpr();
 
+    /**
+     * Holds if the guard directly controls a given basic block. For example in
+     * the following code, the guard `(x > y)` directly controls the block
+     * beneath it:
+     * ```
+     * if (x > y)
+     * {
+     *   Console.WriteLine("x is greater than y");
+     * }
+     * ```
+     * `branch` indicates whether the basic block is entered when the guard
+     * evaluates to `true` or when it evaluates to `false`.
```

Comment:
I'm not sure this explanation is entirely satisfactory, as it doesn't really state what "directly" means. To a lesser degree you could also argue that we never really define the notion of "controls" anywhere. But it might make more sense to postpone fixing these things until we get a more properly shared guards library.

Reply:
I don't have a completely clear picture of what "controls" and "directly" mean myself, in more complex cases (with nested conditions and tested if-then-else constructs, for example).

Reply:
I'd very much like to elaborate on this, but I don't think the signature for the range analysis is the proper place for it.

Reply:
OK, then I suggest we leave the comments as they are (or make minor iterative improvements if any are suggested) and come back to writing clear overall documentation another time.